core/__init__.py

```diff
@@ -115,6 +115,10 @@ def start():
     config = load_config()
     state = None
 
+    #Telemetry dump after x seconds
+    t = threading.Timer(config['core']['dump_interval'], telemetry.dump()) #method: telemetry.dump() not a method right now
```

Comment:
The Timer constructor requires a function as a parameter while you are passing in a function call. Make sure to pass in just `telemetry.dump` instead of `telemetry.dump()`. In addition, for future reference, all threads/timer objects should be using the `partial` function from the `functools` package. 
[Timer Documentation](https://docs.python.org/3/library/threading.html#timer-objects)
[Partial Documentation](https://docs.python.org/3.7/library/functools.html)

Reply:
Recommended Changes: 
- import partial from functools package
  * `from functools import partial`
- replace function call to function object in timer constructor using partials
  * `t = threading.Timer(config['core']['dump_interval'], partial(telemetry.dump))`